src/audio/asrc/asrc_ipc3.c

```diff
@@ -493,170 +207,21 @@ static int asrc_params(struct comp_dev *dev,
-	cd->source_frames = ceil_divide(dev->frames * cd->source_rate,
-					cd->sink_rate);
-
-	/* Use empirical add +10 for target frames number to avoid xruns and
-	 * distorted audio in beginning of streaming. It's slightly more than
-	 * min. needed and does not increase much peak load and buffer memory
-	 * consumption. The copy() function will find the limits and process
-	 * less frames after the buffers levels stabilize.
-	 */
-	cd->source_frames_max = cd->source_frames + 10;
-	cd->sink_frames_max = cd->sink_frames + 10;
-	cd->frames = MAX(cd->source_frames_max, cd->sink_frames_max);
-
-	comp_info(dev, "asrc_params(), source_rate=%u, sink_rate=%u, source_frames_max=%d, sink_frames_max=%d",
-		  cd->source_rate, cd->sink_rate,
-		  cd->source_frames_max, cd->sink_frames_max);
-
-	return 0;
-}
-
-static int asrc_dai_find(struct comp_dev *dev, struct comp_data *cd)
-{
-	struct comp_buffer *sourceb, *sinkb;
-	int pid;
-
-	/* Get current pipeline ID and walk to find the DAI */
-	pid = dev_comp_pipe_id(dev);
-	cd->dai_dev = NULL;
-	if (cd->mode == ASRC_OM_PUSH) {
-		/* In push mode check if sink component is DAI */
-		do {
-			sinkb = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
-
-			dev = sinkb->sink;
-
-			if (!dev) {
-				comp_cl_err(&comp_asrc, "At end, no DAI found.");
-				return -EINVAL;
-			}
-
-			if (dev_comp_pipe_id(dev) != pid) {
-				comp_cl_err(&comp_asrc, "No DAI sink in pipeline.");
-				return -EINVAL;
-			}
-
-		} while (dev_comp_type(dev) != SOF_COMP_DAI);
-	} else {
-		/* In pull mode check if source component is DAI */
-		do {
-			sourceb = list_first_item(&dev->bsource_list, struct comp_buffer,
-						  sink_list);
-
-			dev = sourceb->source;
-
-			if (!dev) {
-				comp_cl_err(&comp_asrc, "At beginning, no DAI found.");
-				return -EINVAL;
-			}
-
-			if (dev_comp_pipe_id(dev) != pid) {
-				comp_cl_err(&comp_asrc, "No DAI source in pipeline.");
-				return -EINVAL;
-			}
-		} while (dev_comp_type(dev) != SOF_COMP_DAI);
-	}
-
-	/* Point dai_dev to found DAI */
-	cd->dai_dev = dev;
-
-	return 0;
-}
-
-static int asrc_dai_configure_timestamp(struct comp_data *cd)
-{
-	if (cd->dai_dev)
-		return cd->dai_dev->drv->ops.dai_ts_config(cd->dai_dev);
-
-	return -EINVAL;
-}
-
-static int asrc_dai_start_timestamp(struct comp_data *cd)
-{
-	if (cd->dai_dev)
-		return cd->dai_dev->drv->ops.dai_ts_start(cd->dai_dev);
-
-	return -EINVAL;
-}
-
-static int asrc_dai_stop_timestamp(struct comp_data *cd)
-{
-	if (cd->dai_dev)
-		return cd->dai_dev->drv->ops.dai_ts_stop(cd->dai_dev);
-
-	return -EINVAL;
-}
-
-#if CONFIG_ZEPHYR_NATIVE_DRIVERS
-	static int asrc_dai_get_timestamp(struct comp_data *cd, struct dai_ts_data *tsd)
-#else
-	static int asrc_dai_get_timestamp(struct comp_data *cd, struct timestamp_data *tsd)
-#endif
-{
-	if (!cd->dai_dev)
-		return -EINVAL;
-
-	return cd->dai_dev->drv->ops.dai_ts_get(cd->dai_dev, tsd);
+	return asrc_param_general(cd, dev, &sourceb->stream, &sinkb->stream);
```

Comment:
I think "common" would be a better description for all these functions - you have several of them